src/hazelcore/extractions/ocaml/OCamlExtraction.re

```diff
@@ -0,0 +1,65 @@
+type t =
+  | OCamlExp(string)
+  | ExtractionFailed(string);
+
+//TODO: test exception handling (seems good now)
+let extract = (ctx: Contexts.t, exp: DHExp.t): t =>
+  switch (OCamlExtraction_Exp.extract(ctx, exp)) {
+  // normal return value
+  | item =>
```

Comment:
If you change `item` to `OCamlExp(exp_str, _exp_typ)`, then you don't need the following `switch`.

Reply:
Fixed.